src/sentry/static/sentry/app/components/sidebar/onboardingStatus.jsx

```diff
@@ -16,6 +17,24 @@ class OnboardingStatus extends React.Component {
     collapsed: PropTypes.bool,
   };
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if (
+      this.props.currentPanel !== nextProps.currentPanel &&
+      (this.props.currentPanel || nextProps.currentPanel == 'todos')
+    ) {
+      this.recordAnalytics(this.props.currentPanel, this.props.org.id);
```

Comment:
This belongs in `componentWillReceiveProps` - `sCU` should be strictly for determining if the component should re-render or not.

Reply:
that makes sense- thanks for explaining why